### DISCUSSION TITLE
How do I get the id of a run that failed?

### DISCUSSION THREAD

**Discussion by davecozz** [community/community#27368] [Actions] [answered]
In a pull request I would like to auto-generate the `Details` link below and provide it in a comment when a build fails, however I cannot determine where the "runs" ID `107433` comes from. 

<img width="935" alt="image" src="https://user-images.githubusercontent.com/7975697/181360246-b720bcec-1c57-435f-8a09-c64014f22c5e.png">

https://github.com/MyOrg/MyRepo/runs/107433?check_suite_focus=true

Does anyone know how to retrieve or calculate this ID?

**Comment by sigprof** [comment, marked as answer]
GitHub does not seem to have a good API for this. Some time ago I hacked up a somewhat working solution, but it's rather complicated and requires adding an unique ID to a step name, which you might not like:

https://github.com/sigprof/test-workflows/blob/e0866f088a12bff34e6f82f45d194a0162efff2f/.github/workflows/get-current-job-info.yml

Note that the code makes the URL for the job output log (and the bare ID in `.id` instead of `.html_url`, if you need that) available only inside the corresponding job — so you can add a step with `if: failure() && !cancelled()` to report failures, but that step must be in the same job as the actual build step which status you want to report.

(I had some more text about that, but it was on github.community and apparently got deleted during the migration :-1: )
